parity/upgrade.rs

```diff
@@ -139,9 +139,7 @@ fn file_exists(path: &Path) -> bool {
 }
 
 pub fn upgrade_key_location(from: &PathBuf, to: &PathBuf) {
-	let mut parent = to.clone();
-	parent.pop();
-	match fs::create_dir_all(&parent).and_then(|()| fs::read_dir(from)) {
+	match fs::create_dir_all(&to).and_then(|()| fs::read_dir(from)) {
```

Comment:
Since the function moves individual files destination directory must be created here.